### PR TITLE
PVCleanup

### PR DESCRIPTION
**Pull request by dwp1084**

### CODE REVIEW

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Seeing it all laid out like this, I think it might be easier to maintain and understand what's going on if the DEV1 and KEV1N classes are completely independent of each other. As it is now, you'd have to jump between PreChargeBase and PreCharge to follow the flow of logic, which is going to make maintaining it a bit of a pain. Ultimately, though, it's up to you whether you want these classes to be separate or together, because you're the person who has to maintain it.

The one other argument I'll give for separating them out is that it will let us merge this branch without doing any additional testing because the main PreCharge code won't have changed. If you do split things up like this, we're going to have to do quite a bit of testing to make sure everything is still working.

No solution is ideal, but I'd like to hear your thoughts.

**Review by mjmagee991 — APPROVED** [review]
Looks great!

File: targets/PreCharge/main.cpp [hunk 1]
```diff
@@ -28,15 +28,15 @@ struct CANInterruptParams {
...
- * NOTE: For this sample, every non-extended (so 11 bit CAN IDs) will be
- * assummed to be intended to be passed as a CANopen message.
- *
- * @param message[in] The passed in CAN message that was read.
- */
+* Interrupt handler to get CAN messages. A function pointer to this function
+* will be passed to the EVT-core CAN interface which will in turn call this
+* function each time a new CAN message comes in.
+*
+* NOTE: For this sample, every non-extended (so 11 bit CAN IDs) will be
```
Comment (aclowmclaughlin) [inline targets/PreCharge/main.cpp:35]: Should you have the "for this sample" notes in here?

File: targets/PreChargeKEV1N/main.cpp [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        if (canStruct != nullptr && canStruct->queue != nullptr) {
+            canStruct->queue->append(message);
+        }
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// CANopen specific Callbacks. Need to be defined in some location
+///////////////////////////////////////////////////////////////////////////////
+extern "C" void CONodeFatalError(void) {
```
Comment (aclowmclaughlin) [inline targets/PreChargeKEV1N/main.cpp:56]: This stuff will be unnecessary eventually with Taylor's CanOpen changes, but I assume it's all good for now.

**Review by aclowmclaughlin — APPROVED** [review]
Looks good, I think!

**Pull request merged**